pennylane/ops/qubit.py

```diff
@@ -3030,6 +3031,43 @@ def diagonalizing_gates(self):
         return []
 
 
+class SparseHamiltonian(Observable):
+    r"""SparseHamiltonian(H)
+    A sparse matrix Hamiltonian observable.
+
+    The expectation value of this observable is
+
+    .. math::
+        E = \langle \psi | H | \psi \rangle
+
+    which is computed directly with a sparse matrix-vector multiplication operation.
```

Comment:
I'm not sure about this part, since it doesn't really describe this class. The equation for the expectation value is true for any observable, and the method for its computation is really a property of the device.

I think it's ok to just remove and keep the simple description above